netlogo-core/src/main/nvm/AnonymousProcedure.scala

```diff
@@ -25,10 +26,11 @@ sealed trait AnonymousProcedure {
   val locals: Array[AnyRef]
   def checkAgentClass(context: Context, agentClassString: String): Unit = {
     val kind = context.agent.kind
-    if (!(((kind == AgentKind.Observer) && agentClassString.contains('O')) ||
-          ((kind == AgentKind.Turtle) && agentClassString.contains('T')) ||
-          ((kind == AgentKind.Patch) && agentClassString.contains('P')) ||
-          ((kind == AgentKind.Link) && agentClassString.contains('L')))) {
+
+    if (!(((kind == AgentKind.Observer) && agentClassString(0) == 'O') ||
```

Comment:
Did you test `kind == AgentKind.Observer` vs `kind eq AgentKind.Observer`? I wouldn't expect it to change the speed in a meaningful way, I was just curious whether it was something you had tried.